File: src/types/device.ts
export type ConnectionType = 'USB' | 'WiFi'

export type DeviceStatus = 'online' | 'offline'

export interface Device {
  id: string
  name: string
  model: string
  osVersion: string
  sn: string
  connectionType: ConnectionType
  status: DeviceStatus
  batteryLevel: number | null
}

File: src-tauri/src/models/device.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Device {
    pub id: String,
    pub name: String,
    pub model: String,
    pub os_version: String,
    pub sn: String,
    pub connection_type: ConnectionType,
    pub status: DeviceStatus,
    pub battery_level: Option<u8>,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum ConnectionType {
    USB,
    WiFi,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum DeviceStatus {
    #[serde(rename = "online")]
    Online,
    #[serde(rename = "offline")]
    Offline,
}
